src/devices/CharacterLcd/ICharacterLcd.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Iot.Device.CharacterLcd
+{
+    /// <summary>
+    /// Interface for character LCD Low-Level handler
+    /// </summary>
+    public interface ICharacterLcd : IDisposable
+    {
+        /// <summary>
+        /// Enable/disable the backlight. (Will always return false if no backlight pin was provided.)
+        /// </summary>
+        bool BacklightOn { get; set; }
+
+        /// <summary>
+        /// Enable/disable the display.
+        /// </summary>
+        bool DisplayOn { get; set; }
+
+        /// <summary>
+        /// Enable/disable the underline cursor.
+        /// </summary>
+        bool UnderlineCursorVisible { get; set; }
+
+        /// <summary>
+        /// Enable/disable the blinking cursor.
+        /// </summary>
+        bool BlinkingCursorVisible { get; set; }
+
+        /// <summary>
+        /// Returns the size of the display.
+        /// </summary>
+        System.Drawing.Size Size { get; }
+
+        /// <summary>
+        /// Returns the number of custom characters for this display.
+        /// A custom character is one that can be user-defined and assigned to a slot using <see cref="CreateCustomCharacter(byte, byte[])"/>
+        /// </summary>
+        int NumberOfCustomCharactersSupported { get; }
```

Comment:
Will all character LCDs suuport custom characters?

Reply:
The ones I saw do, otherwise just set this to 0 in the implementation.